### PR TITLE
cbonsai 1.3.1 (new formula)

### PR DESCRIPTION
**Pull request by mhzawadi**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
 This is take 2, I need some help in getting a test sorted please

### CODE REVIEW

File: Formula/cbonsai.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Cbonsai < Formula
+  desc "Console Bonsai is a bonsai tree generator, written in C using ncurses"
+  homepage "https://github.com/mhzawadi/homebrew-cbonsai"
```
Comment (carlocab) [inline Formula/cbonsai.rb:3]:
```suggestion
  homepage "https://gitlab.com/jallbrit/cbonsai"
```

File: Formula/cbonsai.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  url "https://gitlab.com/jallbrit/cbonsai/-/archive/v1.3.1/cbonsai-v1.3.1.tar.gz"
+  sha256 "62aa7e0eaf3098b7a6a2787146bd2531437df6ad0e604b0f9176128797efd8f9"
+  license "GPL-3.0-only"
+  depends_on "ncurses"
+  depends_on "pkg-config"
```
Comment (carlocab) [inline Formula/cbonsai.rb:8]:
```suggestion
  license "GPL-3.0-only"

  depends_on "pkg-config" => :build
  depends_on "ncurses"
```

File: Formula/cbonsai.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  def install
+    ENV["PKG_CONFIG_PATH"] = "$PKG_CONFIG_PATH:/usr/local/Cellar/ncurses/6.4/lib/pkgconfig/"
+    ENV.append_path "PKG_CONFIG_PATH", "/opt/homebrew/Cellar/ncurses/6.4/lib/pkgconfig/"
```
Comment (carlocab) [inline Formula/cbonsai.rb:12]:
```suggestion
```
`brew` will set this for you automatically.

File: Formula/cbonsai.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    system "make"
+    bin.install "cbonsai"
```
Comment (carlocab) [inline Formula/cbonsai.rb:14]: Why not `make install`?

File: Formula/cbonsai.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+class Cbonsai < Formula
+  desc "Console Bonsai is a bonsai tree generator, written in C using ncurses"
+  homepage "https://github.com/mhzawadi/homebrew-cbonsai"
+  url "https://gitlab.com/jallbrit/cbonsai/-/archive/v1.3.1/cbonsai-v1.3.1.tar.gz"
+  sha256 "62aa7e0eaf3098b7a6a2787146bd2531437df6ad0e604b0f9176128797efd8f9"
+  license "GPL-3.0-only"
```
Comment (carlocab) [inline Formula/cbonsai.rb:6]:
```suggestion
  license "GPL-3.0-or-later"
```
Right?

File: Formula/cbonsai.rb (hunk 4 above)
Comment (mhzawadi) [reply Formula/cbonsai.rb:12]: cuz I dont know how to brew 😢 , will fix. one mo

File: Formula/cbonsai.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  desc "Console Bonsai is a bonsai tree generator, written in C using ncurses"
+  homepage "https://gitlab.com/jallbrit/cbonsai"
+  url "https://gitlab.com/jallbrit/cbonsai/-/archive/v1.3.1/cbonsai-v1.3.1.tar.gz"
+  sha256 "62aa7e0eaf3098b7a6a2787146bd2531437df6ad0e604b0f9176128797efd8f9"
+  license "GPL-3.0-or-later"
+  depends_on "ncurses"
+  depends_on "pkg-config"
+
+  def install
+    system "make install"
```
Comment (carlocab) [inline Formula/cbonsai.rb:11]:
```suggestion
    system "make", "install"
```

File: Formula/cbonsai.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
+class Cbonsai < Formula
+  desc "Console Bonsai is a bonsai tree generator, written in C using ncurses"
+  homepage "https://gitlab.com/jallbrit/cbonsai"
+  url "https://gitlab.com/jallbrit/cbonsai/-/archive/v1.3.1/cbonsai-v1.3.1.tar.gz"
+  sha256 "62aa7e0eaf3098b7a6a2787146bd2531437df6ad0e604b0f9176128797efd8f9"
+  license "GPL-3.0-or-later"
+  depends_on "ncurses"
+  depends_on "pkg-config"
```
Comment (carlocab) [inline Formula/cbonsai.rb:8]:
```suggestion
  license "GPL-3.0-or-later"

  depends_on "pkg-config" => :build
  depends_on "ncurses"
```

File: Formula/cbonsai.rb (hunk 6 above)
Comment (carlocab) [inline Formula/cbonsai.rb:11]:
```suggestion
    system "make", "install"
```

File: Formula/cbonsai.rb [hunk 8]
```diff
@@ -0,0 +1,18 @@
+class Cbonsai < Formula
+  desc "Console Bonsai is a bonsai tree generator, written in C using ncurses"
+  homepage "https://gitlab.com/jallbrit/cbonsai"
+  url "https://gitlab.com/jallbrit/cbonsai/-/archive/v1.3.1/cbonsai-v1.3.1.tar.gz"
+  sha256 "62aa7e0eaf3098b7a6a2787146bd2531437df6ad0e604b0f9176128797efd8f9"
+  license "GPL-3.0-or-later"
+
+  depends_on "pkg-config" => :build
```
Comment (carlocab) [inline Formula/cbonsai.rb:8]:
```suggestion
  depends_on "pkg-config" => :build
  depends_on "scdoc" => :build
```
We need this to use `make install`, it seems.

**Comment by mhzawadi** [comment]
Thanks for you help 👍🏼

**Comment by osalbahr** [comment]
I see that this issue hasn't been merged nor updated for 3 weeks. Are there any issues with it?

**Comment by mhzawadi** [comment]
it failed on MacOS 11, but I dont know how to fix it 😢 as fixing 11 would break 13

`* Dependency 'ncurses' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.`

**Review by carlocab — COMMENTED** [review]
Can you rebase against `origin/master` and then squash all your commits together? I think we can merge this after then (as long as the only CI failure is the complaint about `ncurses`).

**Comment by mhzawadi** [comment]
Hey @carlocab 
Im not that good with git, how do please?

**Comment by carlocab** [comment]
```
cd "$(brew --repository homebrew/core)"
git fetch origin
git checkout mho-cbonsai
git rebase origin/master
git rebase --interactive
```
The last command (`git rebase --interactive`) will open your text editor. For all the lines but the first, change the `pick` to `fixup`, and then quit and save.

When you're done:
```
git push --force
```

**Comment by p-linnane** [comment]
@mhzawadi Please rebase instead of using merge commits.

@carlocab I've noticed once the workflows label gets applied to a PR it's impossible to remove it. Is related to the bottle reuse thing we are trying?

**Comment by mhzawadi** [comment]
this is a big smelly mess, lets try that again 🤦🏼

**Comment by carlocab** [comment]
> @carlocab I've noticed once the workflows label gets applied to a PR it's impossible to remove it. Is related to the bottle reuse thing we are trying?

Yes. Don't remove the https://github.com/Homebrew/homebrew-core/labels/workflows label.

**Comment by osalbahr** [comment]
> this is a big smelly mess, lets try that again 🤦🏼

Wait so a new branch/pr is needed?

**Comment by osalbahr** [comment]
@mhzawadi Any updates? If you're busy I can try to fork your repo and submit a PR

**Comment by mhzawadi** [comment]
> @mhzawadi Any updates? If you're busy I can try to fork your repo and submit a PR

@osalbahr have added you to my homebrew-core fork, if you have the time maybe you could sort the PR?

Its clear I dont have time at the moment 😞

**Comment by osalbahr** [comment]
I tried to create a new PR but I am not sure how to do these steps:

- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

Should I `brew tap mhzawadi/homebrew-core` first?

The instructions in [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md) say to `brew create $URL` but not sure how to do it when a repo already exists. And [Formulae related pull request](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request#formulae-related-pull-request) says to `git remote add <YOUR_USERNAME> https://github.com/<YOUR_USERNAME>/homebrew-core.git` but not sure how it works when the repo is not under my username.

**Comment by osalbahr** [comment]
@mhzawadi FYI:
- #135198

**Pull request closed without merging**